proksee/species_estimator.py

```diff
@@ -161,27 +183,67 @@ def run_mash(self):
 
         LINE_LENGTH_LIMIT = 3500  # Actually 4095, but smaller here for safety.
 
-        output_filepath = os.path.join(self.output_directory, self.OUTPUT_FILENAME)
+        unsorted_output_filepath = os.path.abspath(os.path.join(self.output_directory,
+                                                                self.OUTPUT_FILENAME + ".unsorted"))
+        sorted_output_filepath = os.path.abspath(os.path.join(self.output_directory,
+                                                              self.OUTPUT_FILENAME))
+
+        # Find the common directory between all paths:
+        # This is a safer solution than assuming everything will always use the output directory.
+        common_path = os.path.commonpath(self.input_list)
+
+        if os.path.isdir(common_path):
+            common_directory = common_path
+        else:
+            # The common path is NOT a directory.
+            # This can happen when there is only a single input (the common path is the single file's filepath).
+            # We need to take the absolute path first, because some relative paths (ex: "contigs.fa") will return
+            # an empty string for the dirname.
+            common_directory = os.path.dirname(os.path.abspath(common_path))
 
         # create the mash command
-        command = "mash screen -i 0 -v 1 " + self.mash_database_filename
+        # Use the full file path for the database file:
+        command = ["mash", "screen", "-i", "0", "-v", "1", self.mash_database_filename]
```

Comment:
There is a parameter to mash screen `-p [threads]` which you can use to speed things up.